src/sdl2/event.rs

```diff
@@ -674,6 +676,18 @@ pub enum Event {
         which: u32,
     },
 
+    /// Triggered when the gyroscope or accelerometer is updated
+    ControllerSensorUpdated {
+        timestamp: u32,
+        which: u32,
+        /// The type of the sensor, see SensorType.
+        sensor: i32,
```

Comment:
I think you should directly return SensorType here, or at the very least give a function/method that takes an i32 and returns an `Option<SensorType>`

Reply:
Is it ok to use directly SensorType? SDL may add other sensor types, which would be an invalid enum variant.

On my crate, I use this for enum fields that may be invalid: https://github.com/Yamakaky/joy/blob/master/crates/joycon-sys/src/common.rs#L166

Reply:
I understand that it might be a bit risky to use SensorType directly (we could use the Unknown variant but I see your point), but at the very least we need something to *know* which enum it is from the i32. A `SensorType::from_ll` or `TryFrom<i32> for SensorType` for instance would be a good start. If you already did that impl but I missed it, please let me know.